src/main/java/com/jfrog/ide/common/configuration/JfrogCliDriver.java

```diff
@@ -90,4 +105,57 @@ public CommandResults runCommand(File workingDirectory, String[] args, List<Stri
 
         return commandResults;
     }
+
+    public void downloadCliIfNeeded(String destinationPath, String jfrogCliVersion) throws IOException {
+        // verify installed cli version
+        String cliVersion = extractVersion(version(null));
+        log.debug("Local CLI version is: " + cliVersion);
+        // cli is installed but not the correct version
+        if (cliVersion != null && cliVersion.equals(jfrogCliVersion)) {
+            log.info("Local Jfrog CLI version has been verified and is compatible. Proceeding with its usage.");
+        } else {
+            log.info(String.format("JFrog CLI is either not installed or the current version is incompatible. " +
+                    "Initiating download of version %s to the destination: %s.", jfrogCliVersion, destinationPath));
+            downloadCliFromReleases(jfrogCliVersion, destinationPath);
+        }
+    }
+
+    public void downloadCliFromReleases(String cliVersion, String destinationFolder) throws IOException {
+        String[] urlParts = {"jfrog-cli/v2-jf", cliVersion, "jfrog-cli-" + getOSAndArc(), jfrogExec};
+        String fileLocationInReleases = String.join("/", urlParts);
+        Path basePath = Paths.get(destinationFolder);
+        String destinationPath = basePath.resolve(jfrogExec).toString();
+
+        // download executable from releases and save it in 'destinationPath'
+        try{
+            ServerConfig serverConfig = getServerConfig();
+            ArtifactoryManagerBuilder artifactoryManagerBuilder = createAnonymousAccessArtifactoryManagerBuilder(JFROG_CLI_RELEASES_URL, serverConfig.getProxyConfForTargetUrl(JFROG_CLI_RELEASES_URL), log);
```

Comment:
Please add external remote repo option for aired-gapped using `createArtifactoryManagerBuilder` (`serverConfig.setExternalResourcesRepo`)